R/nicenumericcuts.R

```diff
@@ -0,0 +1,679 @@
+#'      range when using \code{method} of \code{equal.width}.
+#' @param equal.intervals.end A numeric value indicating the end of the
+#'      range when using \code{method} of \code{equal.width}.
+#' @param equal.intervals.increment A numeric value specifying the width of
+#'      the increments when using \code{method} of \code{equal.width}. This
+#'      value is optional, and overrides \code{num.categories} as the number
+#'      of categories is determined using the start, end, and increment.
+#' @param custom.breaks A character containing a comma-seprated list of 
+#'      numeric values to be used as custom break points when the 
+#'      \code{method} is \code{custom}.
+#' @param custom.always.includes.endpoints A logical value indicating whether,
+#'      when \code{method} is \code{custom}, the endpoints are always included.
+#'      If \code{FALSE}, the endpoints for the cuts are determined by the
+#'      \code{custom.breaks} only. 
+#' @param percents A single numeric value, or a character containing a comma
+#'      -separated list of numeric values to be used when the \code{method}
+#'      of \code{percentiles} is used. Values should be between 0 and 100.
+#' @param quantile.type An interger between 1 and 9 to be passed to
+#'      \code{quantile} which determines the algorithm for creating quantiles.
+#' @param factors.use.labels A logical value indicating whether numeric
+#'      information should be extracted from factor labels. If \code{FALSE}
+#'      the function will instead try to extract the underlying numerice
+#'      values from Q/Displayr.
+#' @param grouping.mark A character to be used as the thousands-grouping
+#'      mark when inferring numeric information from factor labels. 
+#' @param decimals.mark A character to be used as the decimals-grouping
+#'      mark when inferring numeric information from factor labels. 
+#'
+#' @return A data frame containing the new factor variables as columns.
+#'
+#' @importFrom flipU ConvertCommaSeparatedStringToVector
+#' @importFrom plyr mapvalues
+#' @importFrom stats var quantile
+#' @export
+NiceNumericCuts <- function(input.data,
+                           method = c("tidy.intervals", "percentiles", "equal.width", "custom"),
+                           num.categories = 2,
+                           right = TRUE,
+                           round.input.data = FALSE,
+                           decimals = 1,
+                           label.decimals = 1,
+                           open.ends = TRUE,
+                           label.style = "tidy.labels",
+                           number.prefix = "",
+                           number.suffix = "",
+                           open.bottom.string = "Less than ",
+                           closed.bottom.string = " and below",
+                           open.top.string = "More than ",
+                           closed.top.string = " and over",
+                           equal.intervals.start = 0,
```

Comment:
```suggestion
                           equal.intervals.start = min(input.data),
```

Reply:
setting NULL default instead as data may not be numeric until coercion.